rqt_topic/src/rqt_topic/topic_widget.py

```diff
@@ -347,9 +347,21 @@ def set_selected_topics(self, selected_topics):
                                                          len(selected_topics)))
         self._selected_topics = selected_topics
 
+    # TODO(Enhancement) Save/Restore tree expansion state
+    def save_settings(self, plugin_settings, instance_settings):
+        header_state = self.topics_tree_widget.header().saveState()
+        instance_settings.set_value('tree_widget_header_state', header_state)
+        instance_settings.set_value('settings_exist', True)
+
+    def restore_settings(self, pluggin_settings, instance_settings):
+        if not instance_settings.value('settings_exist'):
+            return
+        if instance_settings.contains('tree_widget_header_state'):
+            header_state = instance_settings.value('tree_widget_header_state')
+            if not self.topics_tree_widget.header().restoreState(header_state):
+                rospy.logwarn("Failed to restore state, using defaults...")
```

Comment:
Not anymore. It did in debugging stages and it seemed informative to the user if they were like "wtf mate why didn't my column layout reload" to some obscure error.